modules/mono/editor/bindings_generator.cpp

```diff
@@ -1886,6 +1886,96 @@ Error BindingsGenerator::_generate_cs_property(const BindingsGenerator::TypeInte
+		}
+		if ((p_iprop.usage & PropertyUsageFlags::PROPERTY_USAGE_NO_INSTANCE_STATE) != 0) {
+			p_output.append("PropertyUsageFlags.NoInstanceState | ");
+		}
+		if ((p_iprop.usage & PropertyUsageFlags::PROPERTY_USAGE_RESTART_IF_CHANGED) != 0) {
+			p_output.append("PropertyUsageFlags.RestartIfChanged | ");
+		}
+		if ((p_iprop.usage & PropertyUsageFlags::PROPERTY_USAGE_SCRIPT_VARIABLE) != 0) {
+			p_output.append("PropertyUsageFlags.ScriptVariable | ");
+		}
+		if ((p_iprop.usage & PropertyUsageFlags::PROPERTY_USAGE_STORE_IF_NULL) != 0) {
+			p_output.append("PropertyUsageFlags.StoreIfNull | ");
+		}
+		if ((p_iprop.usage & PropertyUsageFlags::PROPERTY_USAGE_UPDATE_ALL_IF_MODIFIED) != 0) {
+			p_output.append("PropertyUsageFlags.UpdateAllIfModified | ");
+		}
+		if ((p_iprop.usage & PropertyUsageFlags::PROPERTY_USAGE_SCRIPT_DEFAULT_VALUE) != 0) {
+			p_output.append("PropertyUsageFlags.ScriptDefaultValue | ");
+		}
+		if ((p_iprop.usage & PropertyUsageFlags::PROPERTY_USAGE_CLASS_IS_ENUM) != 0) {
+			p_output.append("PropertyUsageFlags.ClassIsEnum | ");
+		}
+		if ((p_iprop.usage & PropertyUsageFlags::PROPERTY_USAGE_NIL_IS_VARIANT) != 0) {
+			p_output.append("PropertyUsageFlags.NilIsVariant | ");
+		}
+		if ((p_iprop.usage & PropertyUsageFlags::PROPERTY_USAGE_ARRAY) != 0) {
+			p_output.append("PropertyUsageFlags.Array | ");
+		}
+		if ((p_iprop.usage & PropertyUsageFlags::PROPERTY_USAGE_ALWAYS_DUPLICATE) != 0) {
+			p_output.append("PropertyUsageFlags.AlwaysDuplicate | ");
+		}
+		if ((p_iprop.usage & PropertyUsageFlags::PROPERTY_USAGE_NEVER_DUPLICATE) != 0) {
+			p_output.append("PropertyUsageFlags.NeverDuplicate | ");
+		}
+		if ((p_iprop.usage & PropertyUsageFlags::PROPERTY_USAGE_HIGH_END_GFX) != 0) {
+			p_output.append("PropertyUsageFlags.HighEndGfx | ");
+		}
+		if ((p_iprop.usage & PropertyUsageFlags::PROPERTY_USAGE_NODE_PATH_FROM_SCENE_ROOT) != 0) {
+			p_output.append("PropertyUsageFlags.NodePathFromSceneRoot | ");
+		}
+		if ((p_iprop.usage & PropertyUsageFlags::PROPERTY_USAGE_RESOURCE_NOT_PERSISTENT) != 0) {
+			p_output.append("PropertyUsageFlags.ResourceNotPersistent | ");
+		}
+		if ((p_iprop.usage & PropertyUsageFlags::PROPERTY_USAGE_KEYING_INCREMENTS) != 0) {
+			p_output.append("PropertyUsageFlags.KeyingIncrements | ");
+		}
+		if ((p_iprop.usage & PropertyUsageFlags::PROPERTY_USAGE_DEFERRED_SET_RESOURCE) != 0) {
+			p_output.append("PropertyUsageFlags.DeferredSetResource | ");
+		}
+		if ((p_iprop.usage & PropertyUsageFlags::PROPERTY_USAGE_EDITOR_INSTANTIATE_OBJECT) != 0) {
+			p_output.append("PropertyUsageFlags.EditorInstantiateObject | ");
+		}
+		if ((p_iprop.usage & PropertyUsageFlags::PROPERTY_USAGE_EDITOR_BASIC_SETTING) != 0) {
+			p_output.append("PropertyUsageFlags.EditorBasicSetting | ");
+		}
+		if ((p_iprop.usage & PropertyUsageFlags::PROPERTY_USAGE_READ_ONLY) != 0) {
+			p_output.append("PropertyUsageFlags.ReadOnly | ");
+		}
+		p_output.append("PropertyUsageFlags.None");
```

Comment:
Is there any reason to append this at the end? It's not adding any value as it is a zero

Reply:
It doesn't add the attribute if it is none and I append it because Roslyn will remove it in the c# compile so it doesn't add anything else and it makes it where I don't have to back the or off the end of the string at the end

Reply:
Still redundant no? Why not add a little bit of extra testing and make it clean?

Reply:
Yes it is a bit redundant and you could just instead remove the last 3 char off the string

Reply:
And check for the edge case secret maybe

Reply:
Is this done anywhere else in the source? This reliance on the cleaning up? Does this end up in the generated files, because that seems like a lot of extra redundant text

Reply:
I don't know if secret ever shows up or if it should never show up so it also helps if it can show up

Reply:
I mainly did it for less complexity in the generator because I think that is more important than making the generated files perfect if it comes at no cost to the performance or size of the compiled version